docs/src/app/components/pages/customization/themes.md

```diff
@@ -0,0 +1,229 @@
+We changed how themes work in v0.12.0 (check out
+[release log](https://github.com/callemall/material-ui/releases/tag/v0.12.0) for more details).
+There are now two kinds of themes in Material-UI: **base theme** and **mui theme**.
+The base theme is a plain JS object containing three keys: spacing, palette and fontFamily.
+The mui theme, on the other hand, is a much bigger object. It contains a key for every Material-UI
+component, and the value corresponding to that key describes the styling of that particular component
+under the current base theme. In this sense, the mui theme is *produced* from the base theme.
+The base theme acts as a basis for styling components, whereas the mui theme contains specific values
+(that are calculated based on the base theme) for styling each component.
+
+## Customizing the theme
+
+By default the light base theme is used to calculate the mui theme object. To customize the base theme
+or the calculated mui theme you can use `getMuiTheme` to create a theme object and pass it down the context:
+
+```js
+import React from 'react';
+import Colors from 'material-ui/lib/styles/colors';
+import MuiThemeProvider from 'material-ui/lib/MuiThemeProvider';
+import getMuiTheme from 'material-ui/lib/styles/getMuiTheme';
+import AppBar from 'material-ui/lib/app-bar';
+
+const muiTheme = getMuiTheme({palette: {textColor: Colors.cyan500}});
+
+class Main extends React.Component {
+  render() {
+    return (
+      <MuiThemeProvider muiTheme={muiTheme}>
+        <AppBar title="Hello World!"/>
+      </MuiThemeProvider>
+    );
+  }
+}
+
+export default Main;
+```
+
+Internally, Material-UI components use React's context feature to implement theming. Context is a way
+to pass down values through the component hierarchy without having to use props at every level.
+In fact, context is very convenient for concepts like theming, which are usually implemented in
+a hierarchical manner.
+
+`MuiThemeProvider` uses this feature to pass down the theme to components that need it.
+
+## Using the theme
+
+In case you wish to access the theme object yourself you can use the `muiThemeable` decorator:
+
+```js
+import React from 'react';
+import muiThemeable from 'material-ui/lib/muiThemeable';
+
+class DeepDownTheTree extends React.Component {
+  render() {
+    return (
+      <span style={{color: this.props.muiTheme.baseTheme.palette.textColor}}>
+        Hello World!
+      </span>
+    );
+  }
+}
+
+DeepDownTheTree.propTypes = {
+  muiTheme: React.PropTypes.object,
+};
+
+export default muiThemeable()(DeepDownTheTree);
```

Comment:
so... you mean I should document it's usage as decorator too? O.o

There was documentation for this before. but let's not do that. I think we should keep it simple. I mean they can use this as decorator if they want.